grafana-plugin/src/pages/integrations/Integrations.tsx

```diff
@@ -51,26 +51,31 @@ class Integrations extends React.Component<IntegrationsProps, IntegrationsState>
     errorData: initErrorDataState(),
   };
 
+  private isMounted: boolean;
   private alertReceiveChanneltoPoll: { [key: string]: number } = {};
   private alertReceiveChannelTimerId: ReturnType<typeof setTimeout>;
 
   async componentDidMount() {
+    this.isMounted = false;
     this.update().then(() => this.parseQueryParams(true));
   }
 
+  componentWillUnmount() {
+    this.isMounted = false;
+    clearInterval(this.alertReceiveChannelTimerId);
+  }
+
   componentDidUpdate(prevProps: Readonly<IntegrationsProps>): void {
     if (prevProps.match.params.id && !this.props.match.params.id) {
-      this.setState({ errorData: initErrorDataState() }, () => {
-        this.parseQueryParams();
-      });
+      this.parseQueryParams();
```

Comment:
Removed the initial `setState` as it's doing the same thing inside `parseQueryParams`